clients/include/testing_spmv.hpp

```diff
@@ -0,0 +1,246 @@
+    EXPECT_ROCBLAS_STATUS(rocblas_spmv<T>(nullptr, uplo, N, &alpha, dA, dx, incx, &beta, dy, incy),
+                          rocblas_status_invalid_handle);
+
+    EXPECT_ROCBLAS_STATUS(rocblas_spmv<T>(handle, uplo, N, nullptr, dA, dx, incx, &beta, dy, incy),
+                          rocblas_status_invalid_pointer);
+
+    EXPECT_ROCBLAS_STATUS(
+        rocblas_spmv<T>(handle, uplo, N, &alpha, nullptr, dx, incx, &beta, dy, incy),
+        rocblas_status_invalid_pointer);
+
+    EXPECT_ROCBLAS_STATUS(
+        rocblas_spmv<T>(handle, uplo, N, &alpha, dA, nullptr, incx, &beta, dy, incy),
+        rocblas_status_invalid_pointer);
+
+    EXPECT_ROCBLAS_STATUS(rocblas_spmv<T>(handle, uplo, N, &alpha, dA, dx, incx, nullptr, dy, incy),
+                          rocblas_status_invalid_pointer);
+
+    EXPECT_ROCBLAS_STATUS(
+        rocblas_spmv<T>(handle, uplo, N, &alpha, dA, dx, incx, &beta, nullptr, incy),
+        rocblas_status_invalid_pointer);
+}
+
+template <typename T>
+void testing_spmv(const Arguments& arg)
+{
+    rocblas_int N    = arg.N;
+    rocblas_int incx = arg.incx;
+    rocblas_int incy = arg.incy;
+
+    host_vector<T> alpha(1);
+    host_vector<T> beta(1);
+    alpha[0] = arg.alpha;
+    beta[0]  = arg.beta;
+
+    rocblas_fill uplo = char2rocblas_fill(arg.uplo);
+
+    size_t abs_incx = incx >= 0 ? incx : -incx;
+    size_t abs_incy = incy >= 0 ? incy : -incy;
+
+    size_t size_A = size_t(N) * N;
+    size_t size_X = size_t(N) * abs_incx;
+    size_t size_Y = size_t(N) * abs_incy;
+
+    rocblas_local_handle handle;
+
+    bool bad_uplo = (uplo != rocblas_fill_lower && uplo != rocblas_fill_upper);
+
+    // argument sanity check before allocating invalid memory
+    if(N < 0 || !incx || !incy || bad_uplo)
+    {
+        static const size_t safe_size = 100;
+        device_vector<T>    dA(safe_size);
+        device_vector<T>    dx(safe_size);
+        device_vector<T>    dy(safe_size);
+        if(!dA || !dx || !dy)
+        {
+            CHECK_HIP_ERROR(hipErrorOutOfMemory);
+            return;
+        }
+
+        EXPECT_ROCBLAS_STATUS(rocblas_spmv<T>(handle, uplo, N, alpha, dA, dx, incx, beta, dy, incy),
+                              rocblas_status_invalid_size);
+        return;
+    }
+
+    // Naming: dK is in GPU (device) memory. hK is in CPU (host) memory
+    device_vector<T> d_alpha(1);
+    device_vector<T> d_beta(1);
+
+    host_vector<T> hA(size_A);
+    host_vector<T> hx(size_X);
+    host_vector<T> hy(size_Y);
+    host_vector<T> hy2(size_Y);
+    host_vector<T> hg(size_Y); // gold standard
+
+    double gpu_time_used, cpu_time_used;
+    double rocblas_gflops, cblas_gflops;
+    double h_error, d_error;
+
+    device_vector<T> dA(size_A);
+    device_vector<T> dx(size_X);
+    device_vector<T> dy(size_Y);
+    if(!dA || !dx || !dy)
+    {
+        CHECK_HIP_ERROR(hipErrorOutOfMemory);
+        return;
+    }
+
+    // Initial Data on CPU
+    rocblas_seedrand();
+    rocblas_init_symmetric<T>(hA, N, N);
```

Comment:
Same comment as in sbmv.
Since A is more-or-less a vector here (of size (n * (n+1) / 2)) in the packed version of these algorithms, we can initialize it like a regular vector, we don't have to worry about symmetry, the implementation of spmv handle it correctly either way.